nodejs/eks/userdata.ts

```diff
@@ -211,6 +215,102 @@ ${cfnSignal}
 `;
 }
 
+function createNodeadmUserData(clusterMetadata: ClusterMetadata, args: UserDataArgs): string {
+    if (args.bootstrapExtraArgs) {
+        throw new pulumi.ResourceError(
```

Comment:
TIL pulumi.ResourceError. Should we use it for all validation failures? Does it surface nicely?

Reply:
It associates an error with a resource. I actually never tried using a regular error to see how that gets visualized, because we use `pulumi.ResourceError` throughout the components we have.

But good catch, I missed threading the resource through to here!

Reply:
I was just recently having trouble making validation errors presentable but I didn't notice this one. Maybe it's the answer.

Reply:
I don't think it's the full answer. It still includes the GRPC trace and we cannot anchor them to specific fields

Reply:
Ah ok, sad.